src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,50 @@
+import { useState, useCallback, useEffect } from "react";
+
+export function useLocalStorage<T>(key: string, initialValue: T) {
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    try {
+      const item = window.localStorage.getItem(key);
+      return item ? (JSON.parse(item) as T) : initialValue;
+    } catch (error) {
+      console.error(error);
+      return initialValue;
+    }
+  });
+
+  const setValue = useCallback(
+    (value: T | ((currentValue: T) => T)) => {
+      try {
+        const valueToStore =
+          value instanceof Function ? value(storedValue) : value;
+        setStoredValue(valueToStore);
+        const serializedValue = JSON.stringify(valueToStore);
+        window.localStorage.setItem(key, serializedValue);
+        window.dispatchEvent(
+          new StorageEvent("storage", { key, newValue: serializedValue })
+        );
```

Comment:
This was weird.  Apparently storage events only fire in _other_ open tabs and windows, so to get the update to propagate to all the components int the _current_ window I had to dispatch the event manually (I got this from StackOverflow).

If we had a context provider that was hosting a single instance of the it, they propagation would be more automatic but _most_ components would then a wrapper in their tests.

Reply:
I think this is a good workaround. Context still seems a bit like a heavy-handed approach with how these singletons were being implemented before. Also, I like the idea of not having the components codependent and integrated for such a simple state.